File: tests/expected/julia/generic_functions.jl

function sum_two(x, y)
    return x + y
end

function get_first(container)
    return container[1]
end

function main()
    @assert(sum_two(1, 2) == 3)
    @assert(sum_two(1, 2.0) == 3.0)
    @assert(sum_two("1", "2") == "12")
    @assert(get_first([1, 2, 3]) == 1)
    @assert(get_first(["1", "2", "3"]) == "1")
    @assert(get_first(["1", 2, 3]) == "1")
    @assert(get_first("123") == "1")
end

main()

File: tests/cases/generic_functions.py
from typing import List


def sum_two(x: int, y: int):
    return x + y

def get_first(container: List[int]):
    return container[0]

if __name__ == "__main__":
    assert sum_two(1, 2) == 3
    assert sum_two("1", "2") == "12"
    assert get_first([1,2,3]) == 1
    assert get_first(["1","2","3"]) == "1"
    assert get_first(["1",2,3]) == "1"
    assert get_first("123") == "1"